File: caesar.c

#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{  
    // To check if program was run with one command line.
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    // Convert characters to intergers
    int key = atoi(argv[1]);
    // If user did not input digits in key
    if (key == 0)
     { 
        printf(" Usage: ./caesar key\n");
        return 2; 
     }  
   else
    {
      printf("\n");
    }
    // Prompt user for plaintext
    string s = get_string("Plaintext: ");
    printf("\n");
    
   // Print out ciphertext
    printf("Ciphertext: ");
    
     for(int i = 0; i < strlen (s); i++)
    {
        // To convert string to integer
         int c = (int) s[i];
         
        // To preserve uppercase
         if (c <=97 && c>= 65)    
        {   
         printf("%c", ((c-65 + key)%26) + 65);
        }
        // To preserve lowercase
        else 
        {
             if(c <= 122 && c >= 97)
             {
                printf("%c", (( c-97 + key)%26) + 97);
             }
             // To preserve punctuation
          else
            {
                printf("%c", c);
            }
        }
     }
    
    printf("\n");
}



                   

File: caesar.py
import sys
from cs50 import get_string
from sys import argv

# Check if user input is correct
if len(sys.argv) != 2:
    print("Usage: python caesar.py key")
    exit(1)
try:
    key = int(argv[1])
except:
    print("Usage: python caesar.py key")
    exit(1)
# Get message to be encrypted from user
plaintext = get_string("Plaintext: ")

# Encrypt plain text
plaintext_list = []
for ch in plaintext:
    ascii_plaintext = ord(ch)
    plaintext_list.append(ascii_plaintext)
# Add key
ciphertext_list = []
for i in plaintext_list:
    if i in range(65, 97):
        k = (((i-65+key) % 26)+65)
        ciphertext_list.append(k)
    elif i in range(97, 122):
        j = (((i-97 + key) % 26)+97)
        ciphertext_list.append(j)
    else:
        ciphertext_list.append(i)
ciphertext_charlist = []
for i in ciphertext_list:
    char_ciphertext = chr(i)
    ciphertext_charlist.append(char_ciphertext)
print("ciphertext: ", end="")
# Print cipgertext
print("".join(ciphertext_charlist))
